pgcli/packages/parseutils.py

```diff
@@ -254,6 +256,29 @@ def _parsed_is_open_quote(parsed):
     return False
 
 
+def parse_partial_identifier(word):
+    """Attempt to parse a (partially typed) word as an identifier
+
+    word may include a schema qualification, like `schema_name.partial_name`
+    or `schema_name.` There may also be unclosed quotation marks, like
+    `"schema`, or `schema."partial_name`
+
+    :param word: string representing a (partially complete) identifier
+    :return: sqlparse.sql.Identifier, or None
+    """
+
+    p = sqlparse.parse(word)[0]
+    n_tok = len(p.tokens)
+    if n_tok == 1 and isinstance(p.tokens[0], Identifier):
+        return p.tokens[0]
+    elif (n_tok in (2, 3)) and p.token_next_match(0, Error, '"'):
```

Comment:
Can you elaborate a little bit on what's going on here?

You're checking if the number of tokens is either 2 or 3 and the next match is an Error? 

Why not check if `n_tok` is greater than 1? Is there a specific reason for 2 or 3?